pkg/auth/providers/local/pbkdf2/pbkdf2.go

```diff
@@ -93,26 +95,53 @@ func (p *Pbkdf2) UpdatePassword(userId string, newPassword string) error {
 		return fmt.Errorf("failed to generate salt: %w", err)
 	}
 
-	hashedNewPassword, err := p.hashKey(newPassword, salt, iterations, keyLength)
-	if err != nil {
-		return fmt.Errorf("failed to hash password: %w", err)
-	}
+	var patch []byte
```

Comment:
We can also reuse `hashedNewPassword`
```suggestion
	var patch, hashedNewPassword []byte
```